### ISSUE TITLE
Why is MSAL so difficult to use?

### ISSUE THREAD

**Issue by PaybackMan** [AzureAD/microsoft-authentication-library-for-js#1012]
Can anyone explain why this api is so archaic and hard to use? The amount of confusion in terms of docs, samples, is staggering. We've wasted weeks trying to get a simple Angular app to auth against ad B2C. This shouldn't be a monumental effort! We're about to give up with this approach altogether. Way too many issues for something so simple, we just want to authenticate!

The official samples dont even work FFS

**Comment by negoe** [comment]
Hi @PaybackMan, 

Thanks for reaching out. We are sorry for the inconvenience. I would like to know more about your experience. Were you trying to use MSAL Angular Sample?  We have prioritized MSAL Angular on our Roadmap for coming quarter where we are going to solve existing issues with our past Angular release including the support for Angular 6+. You can reach out to me at neha.goel@microsoft.com. 

Thanks,
Neha

**Comment by chrisgalvan** [comment]
Had the same experience trying to auth against our azure AD, I'm using react and felt like this library should be more mature and abstracted better, there are lots of issues and sometimes just doesn't work, clicking sign-in opens a popup that half the time is blank, not sure what to do to bring it to a working state, tried many things. Looking for other alternatives.

**Comment by sameerag** [comment]
@PaybackMan B2C sample for MSAL JS [here] (https://github.com/Azure-Samples/active-directory-b2c-javascript-msal-singlepageapp). I have added PR #1027 to reflect the change.  We are also working on improving the documentation, sample space and hope it makes it simpler to use this library.

@chrisgalvan  Did you get a chance to check out our react sample [here](https://github.com/AzureAD/microsoft-authentication-library-for-js/tree/dev/samples/react-sample-app). Please let us know (preferably in a separate github issue) on the specifics of the issues you seem to face using MSAL JS with react  and we can help to unblock you. @jasonnutter FYI.

**Comment by chrisgalvan** [comment]
Thanks for pointing me to the samples @sameerag. I didn't realize there were samples in react, I only stumbled across the angular ones.

I think my point is still valid, if you look at the sample you are expecting devs to code [AuthProvider](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/samples/react-sample-app/src/AuthProvider.js) and [auth-utils](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/samples/react-sample-app/src/auth-utils.js) which is kinda a lot to understand. Samples are somewhat basic (which I understand) but we still have to figure out how to integrate with redux or router (which most professional apps will have some sort of state management library and routing).

Are there any plans to release something like AuthProvider wrapper as part of msal (maybe react-msal) library or is the idea that everyone should roll their own?

**Comment by sameerag** [comment]
@chrisgalvan For `Angular` we have plans this quarter to provide a wrapper. `React` is in discussion but we do not have clarity on the dates yet. We have seen requests of both types: folks needing msal-core and want to wrap their own framework of choice around it, and folks asking for specific wrappers. It is a fine balance for us as it needs time and investment to release a wrapper and maintain the code.

Our approach is evolving and could be a mix of both. We will continue supporting `angular` and eventually `react` as they are major asks, for other frameworks like `vue`, `next` etc ... we may just provide a sample going forward.

**Comment by sameerag** [comment]
@PaybackMan Can you please open a new issue(or issues) with clear description/repro steps for all the specific problems you are facing using our library? That would help us move forward. @jasonnutter is working on the Angular wrapper (supporting 'msal-core` later releases and Angular 6+) and he can help address your issues if they are Angular related.

Closing this issue and hoping to hear back from you with specific steps. Thanks.

**Comment by AndrewCraswell** [comment]
@chrisgalvan If you're building a react app and need something to fill the gap before an official wrapper is provided, you can try checking out a package my team maintains unofficially:
https://github.com/syncweek-react-aad/react-aad

This has worked for us, but we've learned that an authentication strategy is often unique to the needs of each individual app. As such, there's a lot of work that could be done to make our wrapper more generic for other's use cases. But at the moment it integrates directly with Redux, and manages the basic authentication process.

Keep in mind though, when an official option becomes available we will probably deprecate the wrapper and transition to something more official.
